File: src/main/java/com/freya02/bot/commands/text/SlashInvite.java
package com.freya02.bot.commands.text;

import com.freya02.botcommands.api.application.ApplicationCommand;
import com.freya02.botcommands.api.application.annotations.Test;
import com.freya02.botcommands.api.application.slash.GuildSlashEvent;
import com.freya02.botcommands.api.application.slash.annotations.JDASlashCommand;

public class SlashInvite extends ApplicationCommand {
	@Test
	@JDASlashCommand(
			name = "invite"
	)
	public void onSlashInvite(GuildSlashEvent event) {
		event.reply(event.getJDA().getInviteUrl()).queue();
	}
}

File: src/main/kotlin/com/freya02/bot/commands/slash/SlashInvite.kt
package com.freya02.bot.commands.slash

import io.github.freya022.botcommands.api.commands.annotations.Command
import io.github.freya022.botcommands.api.commands.application.ApplicationCommand
import io.github.freya022.botcommands.api.commands.application.CommandScope
import io.github.freya022.botcommands.api.commands.application.annotations.Test
import io.github.freya022.botcommands.api.commands.application.slash.GuildSlashEvent
import io.github.freya022.botcommands.api.commands.application.slash.annotations.JDASlashCommand
import io.github.freya022.botcommands.api.commands.application.slash.annotations.TopLevelSlashCommandData

@Command
class SlashInvite : ApplicationCommand() {
    @Test
    @TopLevelSlashCommandData(scope = CommandScope.GUILD)
    @JDASlashCommand(name = "invite")
    fun onSlashInvite(event: GuildSlashEvent) {
        event.reply(event.jda.getInviteUrl()).queue()
    }
}